app/uk/gov/hmrc/economiccrimelevyaccount/controllers/FinancialDataController.scala

```diff
@@ -35,12 +35,24 @@ class FinancialDataController @Inject() (
     extends BackendController(cc)
     with Logging {
 
+  private val loggerContext = "FinancialDataController"
+
   def getFinancialData: Action[AnyContent] = authorise.async { implicit request =>
     integrationFramework
       .getFinancialDetails(request.eclRegistrationReference)
       .map {
-        case Left(errorResponse)  => InternalServerError(Json.toJson(errorResponse))
-        case Right(validResponse) => Ok(Json.toJson(validResponse))
+        case Left(errorResponse)  =>
+          logger.error(
+            s"$loggerContext - Integration Framework error: ${Json.toJson(errorResponse)}" +
```

Comment:
can be done in one go so that we don't have to concatenate the string twice:

`s"$loggerContext - Integration Framework error: ${Json.toJson(errorResponse)} for eclReference ${request.eclRegistrationReference}"`